src/Client.php

```diff
@@ -29,7 +31,7 @@ public static function create(string $token): self
     public static function createWithPsrClient(
         ClientInterface $psrClient,
         string $token,
-    ) {
+    ): self {
```

Comment:
I like the way you designed all this with DI in mind :) 

Only complain I have is that `Client` itself does not have an interface, so it is hard to mock/stub.

Another detail is that `Client` seems to be more like an entry-point: perhaps this class could be called `Notion`:

`$notion = Notion::create(...)->users()->...`

In fact, `users()` gives you a client, while `Notion` could be expanded with further utilities later on